### ISSUE TITLE
Poetry `cleo` not found on Python 3.10 Alpha when using get-poetry.py installation

### ISSUE THREAD

**Issue by thepabloaguilar** [python-poetry/poetry#3828]
- [x] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) Poetry version.
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [ ] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **OS version and name**: Ubuntu (20.04.2)
- **Poetry version**: 1.1.5
- **Link of a [Gist](https://gist.github.com/) with the contents of your pyproject.toml file**: [pyproject.toml](https://github.com/dry-python/returns/blob/master/pyproject.toml)

## Issue

Hello, first of all, thanks for your amazing job!!

I got an error when running poetry on Python 3.10 alpha version.
The reason why I'm using the alpha version in my pipeline is that we want to address possible problems with our library before the official Python release, and we also want to implement some new features in our library using the new features provided by the new version like _Pattern Matching_! (_Disclaimer: We allow failures on Python 3.10 step, so it isn't breaking our pipeline_)

You can find the error output below:
```text
Traceback (most recent call last):
  File "/home/runner/.poetry/bin/poetry", line 17, in <module>
    from poetry.console import main
  File "/home/runner/.poetry/lib/poetry/console/__init__.py", line 1, in <module>
    from .application import Application
  File "/home/runner/.poetry/lib/poetry/console/application.py", line 3, in <module>
    from cleo import Application as BaseApplication
ModuleNotFoundError: No module named 'cleo'
```

For more details, you can also check our pipeline [here](https://github.com/dry-python/returns/runs/2186341189?check_suite_focus=true)

I'd like to know if you can provide some help since Python 3.10 wasn't released officially!

> Btw, maybe it'll be a great idea to enable Python 3.10 in your pipeline too, you just need to allow failure in that step while the official version isn't released yet! For more information [check this out](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idcontinue-on-error)

**Comment by abn** [comment]
Thanks for raising this. This essentially needs a few different changes I believe as I suspect the issue is that `cleo` is skupped during install under `3.10`.

1. Update core to support python releases `>=3.10.0`.
2. Update poetry to use new core release.
2. Update `cleo` metadata and publish a new release.

Nice to haves:
1. Update core (python-poetry/poetry-core#144) and poetry pipelines to add support for 3.10 buids.
2. Maybe remove the need for manual changes when supporting new python releases (this might be ticky, as doing so will loose some metadata).

**Comment by thepabloaguilar** [comment]
Thanks a lot @abn!!
If I can help with something, please, let me know!

**Comment by abn** [comment]
@thepabloaguilar no worries. FYI I misinterpreted the issue since there wasn't sufficient context. See edit above. Additionally, to unblock your `3.10` pipeline, you can see https://github.com/python-poetry/poetry-core/pull/144/commits/81fc0d42d806583096392a822202483e85e1b4e8.

**Comment by abn** [comment]
I have raised https://github.com/python-poetry/poetry/issues/3842 to track CI changes. And since this will be resolved with the new bootstrap method, I will leave this open till that is merged.
